roles/confluent.variables/defaults/main.yml

```diff
@@ -719,3 +719,56 @@ kafka_connect_additional_system_admins: "{{rbac_component_additional_system_admi
 # TODO this one honestly does not make sense. You get rolebindings to otherthings and then c3 allows you to view them. Talk to product
 ### List of users to be granted system admin Role Bindings on the Control Center Cluster
 control_center_additional_system_admins: "{{rbac_component_additional_system_admins}}"
+
+# Secrets Protection variables
+
+### Boolean to enable secrets protection on all components.
+secrets_protection_enabled: false
+
```

Comment:
Include a master key variable please, and then we can have logic that will create a key if there is not one provided

Reply:
this is psuedo code you can add to all.yml
```
---
- name: Generate CA for Self Signed Certs
  hosts: zookeeper[0]
  gather_facts: false
  tags: certificate_authority
  tasks:
  - import_role:
      name: confluent.variables

  - name: Create Certificate Authority and Copy to Ansible Host
    include_tasks: tasks/certificate_authority.yml
    when: (self_signed|bool and regenerate_ca|bool and kafka_broker_listeners | ssl_required(ssl_enabled)) or create_mds_certs|bool

  - name: create master key 
    include_tasks: tasks/masterkey.yml 
    when: when user did not provide a key
```